### ISSUE TITLE
Activity Model

### ISSUE THREAD

**Issue by koellerMC** [cytomine/Cytomine-python-client#56]
Dear Cytomine Team,

I was wondering if the full Activity functionality as mentioned in the [doc](https://doc.uliege.cytomine.org/dev-guide/api/reference#get-/api/project/-project-/annotationaction/count.json) is already integrated in the python client. So far, I was only able to find the AnnotationAction. Has this already been implemented or would this be a feature request?

Thank you very much!
BR
MK

**Comment by koellerMC** [comment]
Dear Cytomine Team, 

I have now written an extension to the client that is able to retrieve `UserConnection` and `UserImageConsultation` data. I have tried to be as consistent with the current framework. It certainly only is a work-around, as it is based on the `Model` class. Potentially the `Collection` class would suit better, however, I did not dig to deep into it. 
The new models follow the same convention in regard to fetching them, and can also be converted to json. 
Currently the retrieved data is stored within `class.collection` as a `list` of `dicts`. 

The following snippet delineates my implementation:

```python
import json

from cytomine.models.model import Model
from cytomine.cytomine import Cytomine

        
class UserConnection(Model):
    def __init__(self):
        super(UserConnection, self).__init__()
        self.project = None
        self.user = None
        self.created = None
        self.deleted = None
        self.id = None
        self.name = None
        self.offset = None
        self.perPage = None
        self.size = None
        self.totalPages = None
        self.updated = None

    def is_new(self):
        return False
    
    def fetch(self, project: int, user: int):
        if not isinstance(project,int) or not isinstance(user,int):
            raise ValueError("Both project and user must be of type int.")
        else:
            self.project = project
            self.user = user

        return Cytomine.get_instance().get_model(self, self.query_parameters)

    def uri(self):
        if self.is_new():
            raise AssertionError ("Currently can not create new instances.")
        else:
            return f'project/{self.project}/userconnection/{self.user}.json'

    def to_json(self, **dump_parameters):
        return json.dumps(self.collection)
    
    
class UserImageConsultation(Model):
    def __init__(self):
        super(UserImageConsultation, self).__init__()
        self.project = None
        self.user = None
        self.created = None
        self.deleted = None
        self.id = None
        self.name = None
        self.offset = None
        self.perPage = None
        self.size = None
        self.totalPages = None
        self.updated = None

    def is_new(self):
        return False
    
    def fetch(self, project: int, user: int):
        if not isinstance(project,int) or not isinstance(user,int):
            raise ValueError("Both project and user must be of type int.")
        else:
            self.project = project
            self.user = user

        return Cytomine.get_instance().get_model(self, self.query_parameters)

    def uri(self):
        if self.is_new():
            raise AssertionError ("Currently can not create new instances.")
        else:
            return f'project/{self.project}/user/{self.user}/imageconsultation.json'

    def to_json(self, **dump_parameters):
        return json.dumps(self.collection)
```

**Comment by koellerMC** [comment]
As a consequence of using the `Model` class the terminal output now displays:

```bash
[2023-06-23 12:24:47,650][INFO] [GET] [userconnection] None : None | 200 OK
```
The `None : None` part might be polished, however, as it works now for our purposes I will implement it.

**Comment by waliens** [comment]
Hi @koellerMC,

Glad you were able to find a workaround for your issue and thanks for sharing it here. An update of the python client is not a priority right now at Cytomine but when we'll come back to it, we'll take a look at your implementation. In the meantime, please feel free to submit a pull request with your changes. 

Have a nice day !

**Comment by jgillardcytomine** [comment]
This is now part of our roadmap. Updates will be provided here.

**Comment by waliens** [comment]
@koellerMC an Activity model equivalent to yours in terms of features has been implemented in version 2.4.0 of the python client. Thank you again for sharing your implementation !

**Comment by koellerMC** [comment]
@waliens thank you very much for the notice! I am looking forward to use it in our upcoming reporting! 

BR
MK
